components/menu-dropdown/menu-dropdown.jsx

```diff
@@ -510,12 +514,8 @@ class MenuDropdown extends React.Component {
 		return undefined;
 	};
 
-	setCurrentSelectedIndices = (nextProps) => {
-		if (this.props.multiple !== true) {
-			this.setState({
-				selectedIndex: this.getIndexByValue(nextProps.value, nextProps.options),
-			});
-		} else {
+	getCurrentSelectedIndices = (nextProps) => {
```

Comment:
Since `setState` cannot be called from the `constructor`, changed this method to only return the state. 

`setState` can be called separately by the method which calls this one.